debug ice: `Span must not be empty and have no suggestion'` : span points at  "where" but is quite off

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.
http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/
-->

### Code
requires debug assertions for ICE
```Rust
extern "C" {
    type Item = [T] where [T]: Sized;
}
```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc 1.72.0-nightly (e6d4725c7 2023-06-05)
binary: rustc
commit-hash: e6d4725c76f3b526c74454bc51afdf6daf133506
commit-date: 2023-06-05
host: x86_64-unknown-linux-gnu
release: 1.72.0-nightly
LLVM version: 16.0.4
```

### Error output

```
error: incorrect `type` inside `extern` block
 --> treereduce.out:2:10
  |
1 | extern "C" {
  | ---------- `extern` blocks define existing foreign types and types inside of them cannot have a body
2 |     type Item = [T] where [T]: Sized;
  |          ^^^^   --- the invalid body
  |          |
  |          cannot have a body
  |
  = note: for more information, visit https://doc.rust-lang.org/std/keyword.extern.html

error: `type`s inside `extern` blocks cannot have `where` clauses
 --> treereduce.out:2:14
  |
1 | extern "C" {
  | ---------- `extern` block begins here
2 |     type Item = [T] where [T]: Sized;
  |              ^ help: remove the `where` clause
  |
  = note: for more information, visit https://doc.rust-lang.org/std/keyword.extern.html

error[E0412]: cannot find type `T` in this scope
 --> treereduce.out:2:28
  |
2 |     type Item = [T] where [T]: Sized;
  |                            ^ not found in this scope

error[E0412]: cannot find type `T` in this scope
 --> treereduce.out:2:18
  |
2 |     type Item = [T] where [T]: Sized;
  |                  ^ not found in this scope

error[E0658]: extern types are experimental
 --> treereduce.out:2:5
  |
2 |     type Item = [T] where [T]: Sized;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #43467 <https://github.com/rust-lang/rust/issues/43467> for more information
  = help: add `#![feature(extern_types)]` to the crate attributes to enable

error[E0601]: `main` function not found in crate `treereduce`
 --> treereduce.out:4:2
  |
4 | }
  |  ^ consider adding a `main` function to `treereduce.out`

error: aborting due to 6 previous errors
```
The problem is the
````
error: `type`s inside `extern` blocks cannot have `where` clauses
 --> treereduce.out:2:14
  |
1 | extern "C" {
  | ---------- `extern` block begins here
2 |     type Item = [T] where [T]: Sized;
  |              ^ help: remove the `where` clause
  |
  = note: for more information, visit https://doc.rust-lang.org/std/keyword.extern.html
````
which is supposed to point at the `where` but it somehow completely misses it and so the point points at nothing particular..?

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```
error: incorrect `type` inside `extern` block
 --> treereduce.out:2:10
  |
1 | extern "C" {
  | ---------- `extern` blocks define existing foreign types and types inside of them cannot have a body
2 |     type Item = [T] where [T]: Sized;
  |          ^^^^   --- the invalid body
  |          |
  |          cannot have a body
  |
  = note: for more information, visit https://doc.rust-lang.org/std/keyword.extern.html

thread 'rustc' panicked at 'Span must not be empty and have no suggestion', /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic.rs:769:9
stack backtrace:
   0:     0x7f019b3631b6 - std::backtrace_rs::backtrace::libunwind::trace::hbf38f71cf3c249c4
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f019b3631b6 - std::backtrace_rs::backtrace::trace_unsynchronized::h92b7b3a804dec4eb
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f019b3631b6 - std::sys_common::backtrace::_print_fmt::h348875c0dc69551d
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x7f019b3631b6 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h9f8410fcd9676ef8
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f019b3b06a8 - core::fmt::rt::Argument::fmt::h4f8c28fcc5e0c6ce
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/fmt/rt.rs:138:9
   5:     0x7f019b3b06a8 - core::fmt::write::h7d46c21d618fb808
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/fmt/mod.rs:1094:21
   6:     0x7f019b31833f - std::io::Write::write_fmt::hdaaa540d9d335de4
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/io/mod.rs:1713:15
   7:     0x7f019b362fb5 - std::sys_common::backtrace::_print::ha4b3dcdc1fa87761
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7f019b362fb5 - std::sys_common::backtrace::print::h2505fe8968f08e73
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7f019b339eac - std::panicking::default_hook::{{closure}}::h6e819d6b421237e7
  10:     0x7f019b339b82 - std::panicking::default_hook::hd033617335ffb85c
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:288:9
  11:     0x7f019de54eeb - <alloc[d322003864c16cf5]::boxed::Box<dyn for<'a, 'b> core[ad2aefeb2646d5e0]::ops::function::Fn<(&'a core[ad2aefeb2646d5e0]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[ad2aefeb2646d5e0]::marker::Sync + core[ad2aefeb2646d5e0]::marker::Send> as core[ad2aefeb2646d5e0]::ops::function::Fn<(&core[ad2aefeb2646d5e0]::panic::panic_info::PanicInfo,)>>::call
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:1999:9
  12:     0x7f019de54eeb - rustc_driver_impl[137e8ec47d2ed4b4]::install_ice_hook::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:1269:13
  13:     0x7f019b33a745 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h031eddff9a16c0f1
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:1999:9
  14:     0x7f019b33a745 - std::panicking::rust_panic_with_hook::h178ca060d6fe37aa
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:709:13
  15:     0x7f019b363481 - std::panicking::begin_panic_handler::{{closure}}::h042578877a3be6e2
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:595:13
  16:     0x7f019b363296 - std::sys_common::backtrace::__rust_end_short_backtrace::hc65e7fa9523157e4
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:151:18
  17:     0x7f019b33a2b2 - rust_begin_unwind
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:593:5
  18:     0x7f019b3c7323 - core::panicking::panic_fmt::hb3d3573ea68cbca7
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panicking.rs:67:14
  19:     0x7f019d38dc91 - <rustc_errors[5695b484f9970627]::diagnostic::Diagnostic>::span_suggestions_with_style::<rustc_error_messages[36385b87b05a1303]::SubdiagnosticMessage, core[ad2aefeb2646d5e0]::array::iter::IntoIter<alloc[d322003864c16cf5]::string::String, 1usize>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic.rs:769:9
  20:     0x7f019d3bb4b0 - <rustc_ast_passes[b35d70ea746bad1b]::errors::ExternTypesCannotHave as rustc_errors[5695b484f9970627]::diagnostic_builder::IntoDiagnostic>::into_diagnostic
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_ast_passes/src/errors.rs:234:10
  21:     0x7f019d3c975c - <rustc_errors[5695b484f9970627]::Handler>::create_err::<rustc_ast_passes[b35d70ea746bad1b]::errors::ExternTypesCannotHave>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/lib.rs:1135:9
  22:     0x7f019d3c975c - <rustc_errors[5695b484f9970627]::Handler>::emit_err::<rustc_ast_passes[b35d70ea746bad1b]::errors::ExternTypesCannotHave>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/lib.rs:1128:14
  23:     0x7f019d3ac8ed - <rustc_ast_passes[b35d70ea746bad1b]::ast_validation::AstValidator>::check_foreign_ty_genericless::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_ast_passes/src/ast_validation.rs:369:13
  24:     0x7f019d3ac8ed - <rustc_ast_passes[b35d70ea746bad1b]::ast_validation::AstValidator>::check_foreign_ty_genericless
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_ast_passes/src/ast_validation.rs:382:13
  25:     0x7f019d3ac8ed - <rustc_ast_passes[b35d70ea746bad1b]::ast_validation::AstValidator as rustc_ast[d85b886c5bb9ebed]::visit::Visitor>::visit_foreign_item
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_ast_passes/src/ast_validation.rs:1042:17
  26:     0x7f019d378b69 - rustc_ast[d85b886c5bb9ebed]::visit::walk_item::<rustc_ast_passes[b35d70ea746bad1b]::ast_validation::AstValidator>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_ast/src/visit.rs:328:13
  27:     0x7f019d3ab0fc - <rustc_ast_passes[b35d70ea746bad1b]::ast_validation::AstValidator as rustc_ast[d85b886c5bb9ebed]::visit::Visitor>::visit_item
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_ast_passes/src/ast_validation.rs:935:17
  28:     0x7f019e9945cd - rustc_ast[d85b886c5bb9ebed]::visit::walk_crate::<rustc_ast_passes[b35d70ea746bad1b]::ast_validation::AstValidator>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_ast/src/visit.rs:269:5
  29:     0x7f019e9945cd - rustc_ast_passes[b35d70ea746bad1b]::ast_validation::check_crate
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_ast_passes/src/ast_validation.rs:1498:5
  30:     0x7f019e9945cd - rustc_interface[bdd613add766aac1]::passes::configure_and_expand::{closure#3}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/passes.rs:274:9
  31:     0x7f019e9945cd - <rustc_data_structures[b02d0c7e382dc253]::profiling::VerboseTimingGuard>::run::<bool, rustc_interface[bdd613add766aac1]::passes::configure_and_expand::{closure#3}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/profiling.rs:752:9
  32:     0x7f019e9945cd - <rustc_session[5a191b6cc5b344a9]::session::Session>::time::<bool, rustc_interface[bdd613add766aac1]::passes::configure_and_expand::{closure#3}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_session/src/utils.rs:11:50
  33:     0x7f019e9945cd - rustc_interface[bdd613add766aac1]::passes::configure_and_expand
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/passes.rs:273:32
  34:     0x7f019e9945cd - rustc_interface[bdd613add766aac1]::passes::resolver_for_lowering
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/passes.rs:571:17
  35:     0x7f019fc0049a - rustc_query_impl[11b733f46d02fd3a]::query_impl::resolver_for_lowering::dynamic_query::{closure#2}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:577:33
  36:     0x7f019fc0049a - rustc_query_impl[11b733f46d02fd3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[11b733f46d02fd3a]::query_impl::resolver_for_lowering::dynamic_query::{closure#2}::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:506:18
  37:     0x7f019fc81088 - rustc_query_impl[11b733f46d02fd3a]::query_impl::resolver_for_lowering::dynamic_query::{closure#2}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:574:25
  38:     0x7f019fc81088 - <rustc_query_impl[11b733f46d02fd3a]::query_impl::resolver_for_lowering::dynamic_query::{closure#2} as core[ad2aefeb2646d5e0]::ops::function::FnOnce<(rustc_middle[d3981eb571cac69d]::ty::context::TyCtxt, ())>>::call_once
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/ops/function.rs:250:5
  39:     0x7f019f95afc5 - <rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false> as rustc_query_system[6750db5a630b1ff]::query::config::QueryConfig<rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:116:9
  40:     0x7f019f95afc5 - rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr::<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:480:72
  41:     0x7f019f95afc5 - rustc_middle[d3981eb571cac69d]::ty::context::tls::enter_context::<rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:82:9
  42:     0x7f019f95afc5 - <std[9f7e6658e0c7b2c6]::thread::local::LocalKey<core[ad2aefeb2646d5e0]::cell::Cell<*const ()>>>::try_with::<rustc_middle[d3981eb571cac69d]::ty::context::tls::enter_context<rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:270:16
  43:     0x7f019f95afc5 - <std[9f7e6658e0c7b2c6]::thread::local::LocalKey<core[ad2aefeb2646d5e0]::cell::Cell<*const ()>>>::with::<rustc_middle[d3981eb571cac69d]::ty::context::tls::enter_context<rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:246:9
  44:     0x7f019f95afc5 - rustc_middle[d3981eb571cac69d]::ty::context::tls::enter_context::<rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:79:9
  45:     0x7f019f95afc5 - <rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt as rustc_query_system[6750db5a630b1ff]::query::QueryContext>::start_query::<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:149:13
  46:     0x7f019f95afc5 - rustc_middle[d3981eb571cac69d]::ty::context::tls::with_related_context::<<rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt as rustc_query_system[6750db5a630b1ff]::query::QueryContext>::start_query<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:133:9
  47:     0x7f019f95afc5 - rustc_middle[d3981eb571cac69d]::ty::context::tls::with_context::<rustc_middle[d3981eb571cac69d]::ty::context::tls::with_related_context<<rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt as rustc_query_system[6750db5a630b1ff]::query::QueryContext>::start_query<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:111:36
  48:     0x7f019f95afc5 - rustc_middle[d3981eb571cac69d]::ty::context::tls::with_context_opt::<rustc_middle[d3981eb571cac69d]::ty::context::tls::with_context<rustc_middle[d3981eb571cac69d]::ty::context::tls::with_related_context<<rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt as rustc_query_system[6750db5a630b1ff]::query::QueryContext>::start_query<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:100:18
  49:     0x7f019f95afc5 - rustc_middle[d3981eb571cac69d]::ty::context::tls::with_context::<rustc_middle[d3981eb571cac69d]::ty::context::tls::with_related_context<<rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt as rustc_query_system[6750db5a630b1ff]::query::QueryContext>::start_query<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:111:5
  50:     0x7f019f95afc5 - rustc_middle[d3981eb571cac69d]::ty::context::tls::with_related_context::<<rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt as rustc_query_system[6750db5a630b1ff]::query::QueryContext>::start_query<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:124:5
  51:     0x7f019f95afc5 - <rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt as rustc_query_system[6750db5a630b1ff]::query::QueryContext>::start_query::<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:134:9
  52:     0x7f019f95afc5 - rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job_non_incr::<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:480:18
  53:     0x7f019f95afc5 - rustc_query_system[6750db5a630b1ff]::query::plumbing::execute_job::<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt, false>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:413:9
  54:     0x7f019f95afc5 - rustc_query_system[6750db5a630b1ff]::query::plumbing::try_execute_query::<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt, false>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:358:13
  55:     0x7f019fb3d01c - rustc_query_system[6750db5a630b1ff]::query::plumbing::get_query_non_incr::<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:810:32
  56:     0x7f019fb3d01c - stacker[dd4b9ba84b899798]::maybe_grow::<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[6750db5a630b1ff]::query::plumbing::get_query_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  57:     0x7f019fb3d01c - rustc_data_structures[b02d0c7e382dc253]::stack::ensure_sufficient_stack::<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[6750db5a630b1ff]::query::plumbing::get_query_non_incr<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  58:     0x7f019fb3d01c - rustc_query_system[6750db5a630b1ff]::query::plumbing::get_query_non_incr::<rustc_query_impl[11b733f46d02fd3a]::DynamicConfig<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[11b733f46d02fd3a]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:810:5
  59:     0x7f019fb3d01c - rustc_query_impl[11b733f46d02fd3a]::query_impl::resolver_for_lowering::get_query_non_incr::__rust_end_short_backtrace
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:554:26
  60:     0x7f019de75c40 - rustc_middle[d3981eb571cac69d]::query::plumbing::query_get_at::<rustc_query_system[6750db5a630b1ff]::query::caches::SingleCache<rustc_middle[d3981eb571cac69d]::query::erase::Erased<[u8; 8usize]>>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:155:17
  61:     0x7f019de75c40 - <rustc_middle[d3981eb571cac69d]::query::plumbing::TyCtxtAt>::resolver_for_lowering
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/mod.rs:2178:1
  62:     0x7f019de75c40 - <rustc_middle[d3981eb571cac69d]::ty::context::TyCtxt>::resolver_for_lowering
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:376:35
  63:     0x7f019de75c40 - rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}::{closure#2}::{closure#2}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:392:52
  64:     0x7f019de75c40 - <rustc_middle[d3981eb571cac69d]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}::{closure#2}::{closure#2}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context.rs:559:37
  65:     0x7f019de75c40 - rustc_middle[d3981eb571cac69d]::ty::context::tls::enter_context::<<rustc_middle[d3981eb571cac69d]::ty::context::GlobalCtxt>::enter<rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}::{closure#2}::{closure#2}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>::{closure#0}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:82:9
  66:     0x7f019de75c40 - <std[9f7e6658e0c7b2c6]::thread::local::LocalKey<core[ad2aefeb2646d5e0]::cell::Cell<*const ()>>>::try_with::<rustc_middle[d3981eb571cac69d]::ty::context::tls::enter_context<<rustc_middle[d3981eb571cac69d]::ty::context::GlobalCtxt>::enter<rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}::{closure#2}::{closure#2}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>::{closure#0}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>::{closure#0}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:270:16
  67:     0x7f019de75c40 - <std[9f7e6658e0c7b2c6]::thread::local::LocalKey<core[ad2aefeb2646d5e0]::cell::Cell<*const ()>>>::with::<rustc_middle[d3981eb571cac69d]::ty::context::tls::enter_context<<rustc_middle[d3981eb571cac69d]::ty::context::GlobalCtxt>::enter<rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}::{closure#2}::{closure#2}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>::{closure#0}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>::{closure#0}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:246:9
  68:     0x7f019de75c40 - rustc_middle[d3981eb571cac69d]::ty::context::tls::enter_context::<<rustc_middle[d3981eb571cac69d]::ty::context::GlobalCtxt>::enter<rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}::{closure#2}::{closure#2}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>::{closure#0}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:79:9
  69:     0x7f019de75c40 - <rustc_middle[d3981eb571cac69d]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}::{closure#2}::{closure#2}, &rustc_data_structures[b02d0c7e382dc253]::steal::Steal<(rustc_middle[d3981eb571cac69d]::ty::ResolverAstLowering, alloc[d322003864c16cf5]::rc::Rc<rustc_ast[d85b886c5bb9ebed]::ast::Crate>)>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context.rs:559:9
  70:     0x7f019dec4ae8 - rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}::{closure#2}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:392:13
  71:     0x7f019dec4ae8 - <rustc_interface[bdd613add766aac1]::interface::Compiler>::enter::<rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}::{closure#2}, core[ad2aefeb2646d5e0]::result::Result<core[ad2aefeb2646d5e0]::option::Option<rustc_interface[bdd613add766aac1]::queries::Linker>, rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/queries.rs:389:19
  72:     0x7f019de741b3 - rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:353:22
  73:     0x7f019de741b3 - rustc_interface[bdd613add766aac1]::interface::run_compiler::<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/interface.rs:332:21
  74:     0x7f019de741b3 - rustc_span[bce242e8e13946ff]::set_source_map::<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_span/src/lib.rs:1034:5
  75:     0x7f019de61c04 - rustc_interface[bdd613add766aac1]::interface::run_compiler::<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/interface.rs:326:13
  76:     0x7f019de61c04 - <scoped_tls[5e59caf9e26823a4]::ScopedKey<rustc_span[bce242e8e13946ff]::SessionGlobals>>::set::<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/scoped-tls-1.0.0/src/lib.rs:137:9
  77:     0x7f019de7c01c - rustc_span[bce242e8e13946ff]::create_session_globals_then::<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_span/src/lib.rs:120:5
  78:     0x7f019de7c01c - rustc_interface[bdd613add766aac1]::util::run_in_thread_pool_with_globals::<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/util.rs:155:38
  79:     0x7f019de7c01c - std[9f7e6658e0c7b2c6]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[bdd613add766aac1]::util::run_in_thread_pool_with_globals<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:135:18
  80:     0x7f019de68a85 - <std[9f7e6658e0c7b2c6]::thread::Builder>::spawn_unchecked_::<rustc_interface[bdd613add766aac1]::util::run_in_thread_pool_with_globals<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#1}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/mod.rs:529:17
  81:     0x7f019de68a85 - <core[ad2aefeb2646d5e0]::panic::unwind_safe::AssertUnwindSafe<<std[9f7e6658e0c7b2c6]::thread::Builder>::spawn_unchecked_<rustc_interface[bdd613add766aac1]::util::run_in_thread_pool_with_globals<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#1}::{closure#0}> as core[ad2aefeb2646d5e0]::ops::function::FnOnce<()>>::call_once
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panic/unwind_safe.rs:271:9
  82:     0x7f019de68a85 - std[9f7e6658e0c7b2c6]::panicking::try::do_call::<core[ad2aefeb2646d5e0]::panic::unwind_safe::AssertUnwindSafe<<std[9f7e6658e0c7b2c6]::thread::Builder>::spawn_unchecked_<rustc_interface[bdd613add766aac1]::util::run_in_thread_pool_with_globals<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#1}::{closure#0}>, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:500:40
  83:     0x7f019de68a85 - std[9f7e6658e0c7b2c6]::panicking::try::<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, core[ad2aefeb2646d5e0]::panic::unwind_safe::AssertUnwindSafe<<std[9f7e6658e0c7b2c6]::thread::Builder>::spawn_unchecked_<rustc_interface[bdd613add766aac1]::util::run_in_thread_pool_with_globals<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:464:19
  84:     0x7f019de68a85 - std[9f7e6658e0c7b2c6]::panic::catch_unwind::<core[ad2aefeb2646d5e0]::panic::unwind_safe::AssertUnwindSafe<<std[9f7e6658e0c7b2c6]::thread::Builder>::spawn_unchecked_<rustc_interface[bdd613add766aac1]::util::run_in_thread_pool_with_globals<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#1}::{closure#0}>, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panic.rs:142:14
  85:     0x7f019de68a85 - <std[9f7e6658e0c7b2c6]::thread::Builder>::spawn_unchecked_::<rustc_interface[bdd613add766aac1]::util::run_in_thread_pool_with_globals<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#1}
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/mod.rs:528:30
  86:     0x7f019de68a85 - <<std[9f7e6658e0c7b2c6]::thread::Builder>::spawn_unchecked_<rustc_interface[bdd613add766aac1]::util::run_in_thread_pool_with_globals<rustc_interface[bdd613add766aac1]::interface::run_compiler<core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>, rustc_driver_impl[137e8ec47d2ed4b4]::run_compiler::{closure#1}>::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ad2aefeb2646d5e0]::result::Result<(), rustc_span[bce242e8e13946ff]::ErrorGuaranteed>>::{closure#1} as core[ad2aefeb2646d5e0]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/ops/function.rs:250:5
  87:     0x7f019b30dcea - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hce91c2cea9ed4f02
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:1985:9
  88:     0x7f019b30dcea - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::he00e15b27167f2ba
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:1985:9
  89:     0x7f019b30c285 - std::sys::unix::thread::Thread::new::thread_start::h5f506801086ccc29
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys/unix/thread.rs:108:17
  90:     0x7f019b49d44b - <unknown>
  91:     0x7f019b520e40 - <unknown>
  92:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.72.0-dev running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [resolver_for_lowering] getting the resolver for lowering
end of query stack
error: aborting due to previous error
```

</p>
</details>

--- a/compiler/rustc_ast_passes/src/ast_validation.rs
+++ b/compiler/rustc_ast_passes/src/ast_validation.rs
@@ -364,7 +364,12 @@ impl<'a> AstValidator<'a> {
         self.err_handler().emit_err(errors::BoundInContext { span, ctx });
     }
 
-    fn check_foreign_ty_genericless(&self, generics: &Generics, where_span: Span) {
+    fn check_foreign_ty_genericless(
+        &self,
+        generics: &Generics,
+        before_where_clause: &TyAliasWhereClause,
+        after_where_clause: &TyAliasWhereClause,
+    ) {
         let cannot_have = |span, descr, remove_descr| {
             self.err_handler().emit_err(errors::ExternTypesCannotHave {
                 span,
@@ -378,9 +383,14 @@ impl<'a> AstValidator<'a> {
             cannot_have(generics.span, "generic parameters", "generic parameters");
         }
 
-        if !generics.where_clause.predicates.is_empty() {
-            cannot_have(where_span, "`where` clauses", "`where` clause");
-        }
+        let check_where_clause = |where_clause: &TyAliasWhereClause| {
+            if let TyAliasWhereClause(true, where_clause_span) = where_clause {
+                cannot_have(*where_clause_span, "`where` clauses", "`where` clause");
+            }
+        };
+
+        check_where_clause(before_where_clause);
+        check_where_clause(after_where_clause);
     }
 
     fn check_foreign_kind_bodyless(&self, ident: Ident, kind: &str, body: Option<Span>) {
@@ -1039,7 +1049,7 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
                 self.check_defaultness(fi.span, *defaultness);
                 self.check_foreign_kind_bodyless(fi.ident, "type", ty.as_ref().map(|b| b.span));
                 self.check_type_no_bounds(bounds, "`extern` blocks");
-                self.check_foreign_ty_genericless(generics, where_clauses.0.1);
+                self.check_foreign_ty_genericless(generics, &where_clauses.0, &where_clauses.1);
                 self.check_foreign_item_ascii_only(fi.ident);
             }
             ForeignItemKind::Static(_, _, body) => {

--- a/tests/ui/extern/issue-112363-extern-item-where-clauses-debug-ice.rs
+++ b/tests/ui/extern/issue-112363-extern-item-where-clauses-debug-ice.rs
@@ -1,0 +1,10 @@
+extern "C" {
+    type Item = [T] where [T]: Sized;
+    //~^ incorrect `type` inside `extern` block
+    //~| `type`s inside `extern` blocks cannot have `where` clauses
+    //~| cannot find type `T` in this scope
+    //~| cannot find type `T` in this scope
+    //~| extern types are experimental
+}
+
+fn main() {}

--- a/tests/ui/extern/issue-112363-extern-item-where-clauses-debug-ice.stderr
+++ b/tests/ui/extern/issue-112363-extern-item-where-clauses-debug-ice.stderr
@@ -1,0 +1,47 @@
+error: incorrect `type` inside `extern` block
+  --> $DIR/issue-112363-extern-item-where-clauses-debug-ice.rs:2:10
+   |
+LL | extern "C" {
+   | ---------- `extern` blocks define existing foreign types and types inside of them cannot have a body
+LL |     type Item = [T] where [T]: Sized;
+   |          ^^^^   --- the invalid body
+   |          |
+   |          cannot have a body
+   |
+   = note: for more information, visit https://doc.rust-lang.org/std/keyword.extern.html
+
+error: `type`s inside `extern` blocks cannot have `where` clauses
+  --> $DIR/issue-112363-extern-item-where-clauses-debug-ice.rs:2:21
+   |
+LL | extern "C" {
+   | ---------- `extern` block begins here
+LL |     type Item = [T] where [T]: Sized;
+   |                     ^^^^^^^^^^^^^^^^ help: remove the `where` clause
+   |
+   = note: for more information, visit https://doc.rust-lang.org/std/keyword.extern.html
+
+error[E0412]: cannot find type `T` in this scope
+  --> $DIR/issue-112363-extern-item-where-clauses-debug-ice.rs:2:28
+   |
+LL |     type Item = [T] where [T]: Sized;
+   |                            ^ not found in this scope
+
+error[E0412]: cannot find type `T` in this scope
+  --> $DIR/issue-112363-extern-item-where-clauses-debug-ice.rs:2:18
+   |
+LL |     type Item = [T] where [T]: Sized;
+   |                  ^ not found in this scope
+
+error[E0658]: extern types are experimental
+  --> $DIR/issue-112363-extern-item-where-clauses-debug-ice.rs:2:5
+   |
+LL |     type Item = [T] where [T]: Sized;
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: see issue #43467 <https://github.com/rust-lang/rust/issues/43467> for more information
+   = help: add `#![feature(extern_types)]` to the crate attributes to enable
+
+error: aborting due to 5 previous errors
+
+Some errors have detailed explanations: E0412, E0658.
+For more information about an error, try `rustc --explain E0412`.

--- a/tests/ui/parser/foreign-ty-semantic-fail.rs
+++ b/tests/ui/parser/foreign-ty-semantic-fail.rs
@@ -15,4 +15,5 @@ extern "C" {
     //~^ ERROR incorrect `type` inside `extern` block
 
     type E: where;
+    //~^ ERROR `type`s inside `extern` blocks cannot have `where` clauses
 }

--- a/tests/ui/parser/foreign-ty-semantic-fail.stderr
+++ b/tests/ui/parser/foreign-ty-semantic-fail.stderr
@@ -61,5 +61,16 @@ LL |     type D = u8;
    |
    = note: for more information, visit https://doc.rust-lang.org/std/keyword.extern.html
 
-error: aborting due to 6 previous errors
+error: `type`s inside `extern` blocks cannot have `where` clauses
+  --> $DIR/foreign-ty-semantic-fail.rs:17:13
+   |
+LL | extern "C" {
+   | ---------- `extern` block begins here
+...
+LL |     type E: where;
+   |             ^^^^^ help: remove the `where` clause
+   |
+   = note: for more information, visit https://doc.rust-lang.org/std/keyword.extern.html
+
+error: aborting due to 7 previous errors
 

